File: series3_handout/2d-FD_pure/plot_fd2d.m
u = load("u_fd.txt");
N = sqrt(length(u));
disp("N=" + N)
u = reshape(u,[N,N])';

clear title xlabel ylabel zlabel
[x,y] = meshgrid(linspace(0,1,N), linspace(0,1,N));
surf(x,y,u);
xlabel('x');
ylabel('y');
zlabel('u');
title("Approximation to the solution")
disp('Press any key to continue...')
pause
cla

% clear title xlabel ylabel zlabel
% surf(x,y, sin(2*pi*x).*cos(2*pi*y))
% xlabel('x');
% ylabel('y');
% zlabel('u');
% disp('Press any key to continue...')
% pause
% cla

clear title xlabel ylabel zlabel
contourf(x,y,abs(u - sin(2*pi*x).*sin(2*pi*y)))
colorbar()
title("Approximation error")
disp('Press any key to continue...')
pause
cla


clear title xlabel ylabel zlabel
n = floor(N/2);
plot(x(n,:), u(n,:));
hold on
plot(x(n,:), sin(2*pi*x(n,:)).*sin(2*pi*y(n,:)));
legend('u', 'exact');
xlabel = 'x';
ylabel = 'u';
title("Cut at y = " + n/(N+1) );

File: series3_handout/2d-FD_pure/plot_fd2d.py
import mpl_toolkits.mplot3d
from numpy import *
from pylab import *

u = loadtxt("u_fd.txt")
N = int(sqrt(u.shape[0]))
print("N=%d" % N)
u.shape = (N,N)

fig = figure()
ax = fig.add_subplot(111, projection='3d')
x,y = mgrid[0:1:N*1j, 0:1:N*1j]
ax.plot_surface(x,y,u)
xlabel('x')
ylabel('y')
#zlabel('u')
title('Approximation to the solution')
show()

contourf(x,y,abs(u - sin(2*pi*x)*sin(2*pi*y)))
colorbar()
xlabel('x')
ylabel('y')
title('Approximation error')
show()

n = int(N/2)
plot(x[:, n], u[:, n], label='u')
plot(x[:,n], sin(2*pi*x[:,n])*sin(2*pi*y[:,n]), label='exact')
xlabel('x')
ylabel('u')
title('Cut at y={}'.format(round(n*1./(N+1), 5)))
legend()
show()
